p2p/rlpx/rlpx.go

```diff
@@ -48,19 +48,45 @@ import (
 // This type is not generally safe for concurrent use, but reading and writing of messages
 // may happen concurrently after the handshake.
 type Conn struct {
-	dialDest  *ecdsa.PublicKey
-	conn      net.Conn
-	handshake *handshakeState
-	snappy    bool
+	dialDest *ecdsa.PublicKey
+	conn     net.Conn
+	session  *sessionState
+
+	// These are the buffers for snappy compression.
+	// Compression is enabled if they are non-nil.
+	snappyReadBuffer  []byte
+	snappyWriteBuffer []byte
```

Comment:
The doc says
```
// This type is not generally safe for concurrent use, but reading and writing of messages
// may happen concurrently after the handshake.
```
If that's indeed the case, then concurrent writes will corrupt these buffers

Reply:
As discussed on Discord, this comment is supposed to mean that Read and Write can be concurrent with each other.